client/ayon_core/pipeline/traits/lifecycle.py

```diff
@@ -0,0 +1,37 @@
+"""Lifecycle traits."""
+from typing import ClassVar
+
+from .trait import TraitBase
+
+
+class Transient(TraitBase):
+    """Transient trait model.
+
+    This model represents a transient trait.
+
+    Attributes:
+        name (str): Trait name.
+        description (str): Trait description.
+        id (str): id should be namespaced trait name with version
+        tags (List[str]): Tags.
+    """
+
+    name: ClassVar[str] = "Transient"
+    description: ClassVar[str] = "Transient Trait Model"
+    id: ClassVar[str] = "ayon.lifecycle.Transient.v1"
+
+
+class Persistent(TraitBase):
+    """Persistent trait model.
+
+    This model represents a persistent trait.
+
+    Attributes:
+        name (str): Trait name.
+        description (str): Trait description.
+        id (str): id should be namespaced trait name with version
+    """
```

Comment:
Can we improve the docstring to describe the differences between these and how/why to use them?

Reply:
Definitely, that is still in the progress once I am sure I have most needed. There are some Traits that won't be needed probably.